### PR TITLE
Fully annotate `grid.py`

### PR DESCRIPTION
**Pull request by user27182**
### Overview

Add annotations.

### CODE REVIEW

File: pyvista/core/grid.py [hunk 1]
```diff
@@ -685,7 +694,7 @@ def points(self) -> NumpyArray[float]:
         return points
 
     @points.setter
-    def points(self, points):  # numpydoc ignore=PR01
+    def points(self: ImageData, points: Never) -> Never:  # numpydoc ignore=PR01
```
Comment (tkoyama010) [inline pyvista/core/grid.py:697]: I did not know about the type `Never` . Please wait as I am learning its usage.

**Review by tkoyama010 — CHANGES_REQUESTED** [review]

File: pyvista/core/grid.py [hunk 2]
```diff
@@ -214,20 +221,20 @@ def __init__(
             else:
                 raise TypeError('Arguments not understood by `RectilinearGrid`.')
 
-    def __repr__(self):
+    def __repr__(self: RectilinearGrid) -> str:
         """Return the default representation."""
         return DataSet.__repr__(self)
 
-    def __str__(self):
+    def __str__(self: RectilinearGrid) -> str:
```
Comment (user27182) [inline pyvista/core/grid.py:228]: Maybe use Self here

File: pyvista/core/grid.py (hunk 1 above)
Comment (user27182) [reply pyvista/core/grid.py]: I think it has an alias `NoReturn` which may be more explicit.

We can also remove it and hint the actual types, i.e. make them have the same signature as `PointSet.points` setter

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**